src/rgw/rgw_sal_motr.cc

```diff
@@ -4608,21 +4605,19 @@ int MotrMultipartWriter::complete(size_t accounted_size, const std::string& etag
   encode(attrs, bl);
   part_obj->meta.encode(bl);
 
-  string p = "part.";
-  char buf[32];
-  snprintf(buf, sizeof(buf), "%08d", (int)part_num);
-  p.append(buf);
-  string tenant_bkt_name = get_bucket_name(head_obj->get_bucket()->get_tenant(), head_obj->get_bucket()->get_name());
   //This is a MultipartComplete operation so this should always have valid upload id.
-  string upload_id_str = upload_id;
-  string obj_part_iname = "motr.rgw.object." + tenant_bkt_name + "." +
-	                  head_obj->get_key().to_str() + "." + upload_id_str + ".parts";
-  ldpp_dout(dpp, 20) <<__func__ << ": object part index=" << obj_part_iname << dendl;
+  string part = head_obj->get_key().to_str() + "." + upload_id;
```

Comment:
`get_key.to_str()` will in this case return `obj_name` since version_id is not yet generated, but in case version_id is present it will return `obj_name[version_id]`. _(this will never happen but function suggests we want such formatting)_
also we should not use `get_key().to_str()` as we have `get_key_str()` which has different structure i.e `obj1[v1] vs obj1^Gv1`
```suggestion
  string part = head_obj->get_name() + "." + upload_id;
```

Reply:
Very good catch. Changed to `get_key_str()` here and in several other places. Please check. Thanks!

Reply:
No no, we will ned `get->name()` mostly.
The problem is, in `part_upload` i.e. in `MotrMultipartWriter::complete()`, we don't have the version_id set yet, so `get_key_str()` will return `obj_name + '\a'`.
But After `MotrMultipartUpload::complete()` we will have version_id stored in the metadata, so in delete flow i.e in `remove_mobj_and_index_entry()`, `get_key_str()` will return `obj_name + '\a' + version_id`.

Reply:
Right, I realised this by myself also. Fixed again to `->get_name()`. Thanks!